libraries/abstractions/secure_sockets/test/iot_test_tcp.c

```diff
@@ -3053,7 +3059,7 @@ static void prvThreadSafeDifferentSocketsDifferentTasks( void * pvParameters )
                                           ipconfigTCP_MSS - xReceivedBytes,        /* The size of the buffer provided to receive the data. */
                                           0 );                                     /* No flags. */
 
-                if( xReturned <= 0 )
+                if( xReturned < 0 )
```

Comment:
If `xReturned` is always 0, is it possible to be stuck in this loop forever? Should we also check for a timeout here the same way we do in the other tests?

Reply:
xReturned should not always get 0, because it will receive data from the echo server. Considering the received data could be send back from a remote server, which is not guaranteed latency, it should be expected that some SOCKETS_Recv will timeout and received xReturned = 0. But it should not get SOCKETS_Recv = 0 every time. We can add a check for prvCheckTimeout, but this test case seems not checking that oriented, but we can add it if we decide to.

Reply:
It won't always return 0 but what will happen when the echo server stops returning any data from the payload? It is possible to create a rogue echo server that will cause an infinite loop in this test to occur. It is a corner case, but I think we have to account for that IMO.

Reply:
We discussed in chat that a potential infinite loop for these tests specifically is probably not much of a concern.